### DISCUSSION TITLE
@wagmi/core V2: How to use watchAccount to update my data.

### DISCUSSION THREAD

**Discussion by Nilbristi** [wevm/wagmi#3414] [Q&A] [answered]
I am using @wagmi/core V2 with Angular 16.
In migration doc they are suggesting to use **watchAccount** instead of **watchNetwork**

I have used this code but not able to update my variable inside onChange method. How can I do it?
I am getting the console data and now I want to use it. Like want to change my wallet address or chain id and so on.

```
const unwatch = watchAccount(config, {
      onChange(data) {
        console.log('Account changed!', data);
        <I want to call some function here or pass this data or update some variable>
      }
    });
```

**Comment by Nilbristi** [comment]
@here IS there anyone can help me out?

**Reply by Rubanrubi** [reply, marked as answer]
@Nilbristi 

 this.unwatch = watchAccount(wagmiConfig, {
      onChange: (data) => {
        console.log('Account changed!', data);
        if (data.address != undefined) {
              this.getUser(data.address);
            }
        const chains = wagmiConfig.chains
        const chain = chains.find(chain => chain.id === data.chainId)

      },
    })
    
    you can change OnChange function to arrow function it works

**Reply by Nilbristi** [reply]
Thank you @Rubanrubi
